### PR TITLE
Tx.go updates / simplify Validate func

### PR DESCRIPTION
**Pull request by florent-uzio**
# Title

This PR aims to remove some unused func from `tx.go`: `UnmarshallTx` and remove the use of that func in some tests files.
It also simplifies the `Validate` function as there is no need to check if a field is a string or not as it is enforced by Go types.

Tx.go has 100% test coverage.
I will update in a later PR the coverage for `AMMBid` which seems to have decreased.

image

### CODE REVIEW

**Review by GuillemGarciaDev — CHANGES_REQUESTED** [review]
I would also remove the `UnmarshalTx` method on every transaction. Good PR!

**Comment by florent-uzio** [comment]
@GuillemGarciaDev I removed the other `UnmarshallJSON` function from the remaning transaction types and the test files. It's ready for another review

File: xrpl/transaction/tx.go
Comment (GuillemGarciaDev) [inline xrpl/transaction/tx.go:1]: Why  are the BaseTx field validations on `Validate` method being deleted? 🤔

**Review by GuillemGarciaDev — CHANGES_REQUESTED** [review]

File: xrpl/transaction/tx.go
Comment (florent-uzio) [reply xrpl/transaction/tx.go:1]: I originally looked at `xrpl.js` for the validation. And I forgot the fact that `xrpl.js` can be used with Javascript (not only typescript) which allows to define whatever type of field. So it is important there to check that Sequence is a number, Fee is a string etc.
But Go has types and it is not possible to define whatever we want for a field. This is why I removed the validations.
Does that make sense?

Comment (GuillemGarciaDev) [reply xrpl/transaction/tx.go:1]: Yes, but field value validation should be also done. We should validate that fields like `Fee` have a correct numeric value despite their are of type string, and so on. Can you restore the changes and add these validations? 🙏🏻

**Review by GuillemGarciaDev — APPROVED** [review]

**Pull request merged**